tuf/client_rework/updater_rework.py

```diff
@@ -195,318 +196,111 @@ def download_target(
-        timestamp_url = parse.urljoin(self._metadata_base_url, "timestamp.json")
-        data = download.download_bytes(
-            timestamp_url,
-            settings.DEFAULT_TIMESTAMP_REQUIRED_LENGTH,
-            self._fetcher,
-            strict_required_length=False,
-        )
-        self._metadata["timestamp"] = self._verify_timestamp(data)
-        self._metadata["timestamp"].persist(
-            self._get_full_meta_name("timestamp.json")
+        """Load local and remote timestamp metadata"""
+        try:
+            data = self._load_local_metadata("timestamp")
+            self._bundle.update_timestamp(data)
+        except (OSError, exceptions.RepositoryError) as e:
+            # Local load can fail: it's not fatal
+            logger.debug("Failed to load local timestamp %s", e)
+
+        # Load from remote (whether local load succeeded or not)
+        data = self._download_metadata(
+            "timestamp", DEFAULT_TIMESTAMP_MAX_LENGTH
         )
+        self._bundle.update_timestamp(data)
+        self._persist_metadata("timestamp", data)
 
     def _load_snapshot(self) -> None:
-        """
-        TODO
-        """
+        """Load local (and if needed remote) snapshot metadata"""
         try:
-            length = self._metadata["timestamp"].snapshot["length"]
-        except KeyError:
-            length = settings.DEFAULT_SNAPSHOT_REQUIRED_LENGTH
-
-        # Uncomment when implementing consistent_snapshot
-        # if self._consistent_snapshot:
-        #     version = self._metadata["timestamp"].snapshot["version"]
-        # else:
-        #     version = None
-
-        # TODO: Check if exists locally
-        snapshot_url = parse.urljoin(self._metadata_base_url, "snapshot.json")
-        data = download.download_bytes(
-            snapshot_url,
-            length,
-            self._fetcher,
-            strict_required_length=False,
-        )
-
-        self._metadata["snapshot"] = self._verify_snapshot(data)
-        self._metadata["snapshot"].persist(
-            self._get_full_meta_name("snapshot.json")
-        )
-
-    def _load_targets(self, targets_role: str, parent_role: str) -> None:
-        """
-        TODO
-        """
+            data = self._load_local_metadata("snapshot")
+            self._bundle.update_snapshot(data)
+        except (OSError, exceptions.RepositoryError) as e:
+            # Local load failed: we must update from remote
+            logger.debug("Failed to load local snapshot %s", e)
+
+            metainfo = self._bundle.timestamp.signed.meta["snapshot.json"]
+            length = metainfo.get("length") or DEFAULT_SNAPSHOT_MAX_LENGTH
+            version = None
+            if self._bundle.root.signed.consistent_snapshot:
+                version = metainfo["version"]
+
+            data = self._download_metadata("snapshot", length, version)
+            self._bundle.update_snapshot(data)
+            self._persist_metadata("snapshot", data)
```

Comment:
I was confused at first by the downloading of new metadata all happening in the `except` block. It took me a second read to realise that the `_bundle.update_snapshot()` call fails if the snapshot version is out of date with a `RepositoryError`, and then the logic beneath the `except` clause is executed.

Reply:
yeah, basically the only case where we don't want to download snapshot is if we have a local snapshot file AND it is valid -- but with exceptions programming this is what it looks like...

Better comments wouldn't hurt